File: InsuraTech.Domain/Common/AggregateRoot.cs
using MediatR;

namespace InsuraTech.Domain.Common
{
    public abstract class AggregateRoot : Entity
    {
        private readonly List<IDomainEvent> _domainEvents = new();
        public IReadOnlyCollection<IDomainEvent> DomainEvents => _domainEvents.AsReadOnly();
        protected void AddDomainEvent(IDomainEvent domainEvent) =>
            _domainEvents.Add(domainEvent);
        public void ClearDomainEvent() => _domainEvents.Clear();
    }

    public interface IDomainEvent : INotification
    {
        Guid Id { get; }
        Guid EventId { get; }
        DateTime OccurredOn { get; }
        string EventType { get; }
    }
}

File: backend_v2/src/main/java/com/insuratech/domain/common/AggregateRoot.java
package com.insuratech.domain.common;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public abstract class AggregateRoot extends Entity {

    private final List<IDomainEvent> domainEvents = new ArrayList<>();

    protected AggregateRoot() {
        super();
    }

    protected void addDomainEvent(IDomainEvent event) {
        domainEvents.add(event);
    }

    public List<IDomainEvent> getDomainEvents() {
        return Collections.unmodifiableList(domainEvents);
    }

    public void clearDomainEvents() {
        domainEvents.clear();
    }
}
